applications/tari_validator_node/src/p2p/services/epoch_manager/base_layer_epoch_manager.rs

```diff
@@ -232,17 +237,18 @@ impl BaseLayerEpochManager {
     }
 
     pub fn current_epoch(&self) -> Epoch {
-        // let tip = self
-        //     .base_node_client
-        //     .clone()
-        //     .get_tip_info()
-        //     .await
-        //     .unwrap()
-        //     .height_of_longest_chain;
-        // Epoch(tip - 100)
         self.current_epoch
     }
 
+    pub async fn get_shard_id(&mut self, epoch: Epoch, public_key: &PublicKey) -> Result<ShardId, EpochManagerError> {
+        let shard_id = self
```

Comment:
This could be incredibly heavy. Should we not replicate the logic here locally?

Reply:
The base node leverages an index and LMDB cursors and searches just over the validity period for the epoch. So I don't think it'll be too slow, and we could index it directly to make it super fast. TODO I think we should scan and index validator node registrations in the VN